src/movies.js

```diff
@@ -1,31 +1,141 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+function getAllDirectors(ListOfMovies) {
+  let ListOfDir = ListOfMovies.map(function (element) {
+    return element['director'];
+  });
+
+  return ListOfDir;
+}
+// console.log('---------');
+// console.log('---------');
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+function howManyMovies(Datalist) {
+  // we are going to find out how amny drama movies spielberg made
+  // we need to sort out all the movies speilberg made / map
+  // than we need to filter wich one of them are drama / filter
+  let listOfSpielberg = Datalist.filter(function (element) {
+    return element['director'] === 'Steven Spielberg';
+    // This is an element
+
+    // {
+    //   title: 'Indiana Jones and the Last Crusade',
+    //   year: 1989,
+    //   director: 'Steven Spielberg',
+    //   duration: '2h 7min',
+    //   genre: ['Action', 'Adventure', 'Fantasy'],
+    //   score: 8.3
+    // },
+  });
+  let listOfDrama = listOfSpielberg.filter(function (element) {
+    for (let i = 0; i < element['genre'].length; i++) {
+      let current_genre = element['genre'][i];
+      //return current_genre === "Drama";
+      if (current_genre === 'Drama') {
+        return true;
+      }
+    }
+    return false;
+  });
```

Comment:
to filter only drama movies, you can use exact same syntax as above for director.  Since `filter() `method is already  a loop, you do not need to as additional` for loop`